File: python/exercises/happiness/dslc_documentation/functions/clean_happiness.py


# a function for cleaning the world happiness dataset
def clean_happiness(happiness_orig, predictor_variable = None):
  # rename column names
  happiness_clean = happiness_orig.rename(columns={
    "Life Ladder": "happiness",
    "Log GDP per capita": "log_gdp_per_capita",
    "Social support": "social_support",
    "Healthy life expectancy at birth": "life_expectancy",
    "Freedom to make life choices": "freedom_choices",
    "Generosity": "generosity",
    "Perceptions of corruption": "corruption",
    "Positive affect": "positive_affect",
    "Negative affect": "negative_affect", 
    "Confidence in national government": "government_confidence",
    "gini of household income reported in Gallup, by wp5-year": "gini_index"})
  # filter to relevant columns
  happiness_clean = happiness_clean[["country", "year", "happiness", "log_gdp_per_capita",
                                     "social_support", "life_expectancy",
                                     "freedom_choices", "generosity",
                                     "corruption", "positive_affect", 
                                     "negative_affect", "government_confidence",
                                     "gini_index"]]
  
  if (predictor_variable is not None): 
    happiness_clean = happiness_clean[["country", "year", "happiness", predictor_variable]]
  
  return(happiness_clean)

File: R/exercises/happiness/dslc_documentation/functions/cleanHappiness.R


# a function for cleaning the world happiness dataset
cleanHappiness <- function(happiness_orig,
                           .predictor_variable = NULL) {
  happiness_clean <- happiness_orig %>%
    select(country = country, 
           year = year,
           happiness = `Life Ladder`,
           log_gdp_per_capita = `Log GDP per capita`,
           social_support = `Social support`,
           life_expectancy = `Healthy life expectancy at birth`,
           freedom_choices = `Freedom to make life choices`,
           generosity = Generosity,
           corruption = `Perceptions of corruption`,
           positive_affect = `Positive affect`,
           negative_affect = `Negative affect`,
           government_confidence = `Confidence in national government`,
           gini_index = `gini of household income reported in Gallup, by wp5-year`)
  
  if (!is.null(.predictor_variable)) {
    happiness_clean <- happiness_clean %>%
      select(all_of(c("country", "year", "happiness", .predictor_variable)))
  }
  return(happiness_clean)
}
